### DISCUSSION TITLE
Building ONNXruntime for inferencing - Android (armv8) on Windows/Linux fails: "unknown type name 'bfloat16_t'"

### DISCUSSION THREAD

**Discussion by fanelaru** [microsoft/onnxruntime#19645] [Other Q&A] [answered]
I tried on both Windows and Linux to build Android ONNXRuntime for inference for ARM and both of them fails with this error:

`onnxruntime/core/mlas/lib/mlasi.h:366:11: error: unknown type name 'bfloat16_t'; did you mean 'float16_t'?`

I followed the instructions from here: https://onnxruntime.ai/docs/build/android.html

The command:
`build.bat --config Release --parallel --android --android_sdk_path "C:\Android" --android_ndk_path "C:\Android\ndk\26.2.11394342" --android_abi arm64-v8a --android_api 33 --cmake_generator Ninja`

I see in CI, the build for Android is successful but I noticed that `android_abi=x86_64` and not `arm64-v8a`.

**Comment by fanelaru** [comment, marked as answer]
Damn! `--arm64` was missing from my command line. I noticed `build_arm64.bat` file and I looked in to `build.py`

**Reply by siy415** [reply]
Hello! My code occurs same issue now. Could you explain how to fix this issue? Below scipt is mine but I don't know what difference with yours.

My build envrionment
Cmake : 3.29.3
python : 3.8.10
ubuntu : 20.04.6 LTS

``` bash
./build.sh --config Release \
--include_ops_by_config=./tools/ci_build/github/android/mobile_package.required_operators.config \
--android \
--parallel \
--cmake_generator=Ninja \
--build_java \
--build_shared_lib \
--minimal_build=extended \
--disable_rtti \
--disable_ml_ops \
--disable_exceptions \
--enable_reduced_operator_type_support \
--use_nnapi \
--skip_tests \
--android_abi=arm64-v8a \
--skip_submodule_sync \
--android_ndk_path /home/Android/Sdk/ndk/23.1.7779620/
```
